Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -289,69 +344,135 @@ void GameList::OpenWiki()
+        (decompress && file->GetBlobType() != DiscIO::BlobType::GCZ) ||
+        (!decompress && file->GetBlobType() != DiscIO::BlobType::PLAIN))
+    {
+      it.remove();
+      continue;
+    }
+
+    if (!wii_warning_given && !decompress && file->GetPlatform() == DiscIO::Platform::WiiDisc)
+    {
+      QMessageBox wii_warning(this);
+      wii_warning.setIcon(QMessageBox::Warning);
+      wii_warning.setText(tr("Are you sure?"));
+      wii_warning.setInformativeText(tr(
+          "Compressing a Wii disc image will irreversibly change the compressed copy by removing "
+          "padding data. Your disc image will still work. Continue?"));
+      wii_warning.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+
+      if (wii_warning.exec() == QMessageBox::No)
+        return;
+
+      wii_warning_given = true;
+    }
+  }
 
-  QProgressDialog progress_dialog(compressed ? tr("Decompressing...") : tr("Compressing..."),
-                                  tr("Abort"), 0, 100, this);
-  progress_dialog.setWindowModality(Qt::WindowModal);
+  if (files.size() == 0)
+    return;  // We shouldn't get here normally...
 
-  bool good;
+  QString dst_dir;
+  QString dst_path;
 
-  if (compressed)
+  if (files.size() > 1)
   {
-    good = DiscIO::DecompressBlobToFile(original_path, dst_path.toStdString(), &CompressCB,
-                                        &progress_dialog);
+    dst_dir = QFileDialog::getExistingDirectory(
+        this,
+        decompress ? tr("Select where you want to save the decompressed images") :
+                     tr("Select where you want to save the compressed images"),
+        QFileInfo(QString::fromStdString(GetSelectedGame()->GetFilePath())).dir().absolutePath());
+
+    if (dst_dir.isEmpty())
+      return;
   }
   else
   {
-    good = DiscIO::CompressFileToBlob(original_path, dst_path.toStdString(),
-                                      file->GetPlatform() == DiscIO::Platform::WiiDisc ? 1 : 0,
-                                      16384, &CompressCB, &progress_dialog);
+    dst_path = QFileDialog::getSaveFileName(
+        this,
+        decompress ? tr("Select where you want to save the decompressed image") :
+                     tr("Select where you want to save the compressed image"),
+        QFileInfo(QString::fromStdString(GetSelectedGame()->GetFilePath()))
+            .dir()
+            .absoluteFilePath(
+                QFileInfo(QString::fromStdString(files[0]->GetFilePath())).completeBaseName())
+            .append(decompress ? QStringLiteral(".gcm") : QStringLiteral(".gcz")),
+        decompress ? tr("Uncompressed GC/Wii images (*.iso *.gcm)") :
+                     tr("Compressed GC/Wii images (*.gcz)"));
   }
 
-  if (good)
-  {
-    QMessageBox(QMessageBox::Information, tr("Success!"), tr("Successfully compressed image."),
-                QMessageBox::Ok, this)
-        .exec();
-  }
-  else
+  for (int i = 0; i < files.size(); i++)
```

Comment:
This can be a range-based for loop: `for (const auto& file : files)`

Reply:
Ah, thanks, I wasn't aware C++ had that kind of syntax.